PCF-ADR-0003-codeflare-release-process.md

```diff
@@ -0,0 +1,123 @@
+# Standardized release process for Project CodeFlare
+
+|                |                          |
+| -------------- | ------------------------ |
+| Date           | 04/28/2023               |
+| Scope          |                          |
+| Status         | Proposed                    |
+| Authors        | [Anish Asthana](@anishasthana) [Mustafa Eyceoz](@Maxusmusti) |
+| Supersedes     | N/A                      |
+| Superseded by: | N/A                      |
+| Issues         |                          |
+| Other docs:    | PCF-ADR-002              |
+
+## What
+
+This ADR introduces a unified release process, testing, and versioning scheme for Project CodeFlare.
+[PCF-ADR-002](./PCF-ADR-0002-release-story-and-branch-maintenance.md) is a related ADR to this one but is purely supplemental. ADR-002 covers the branching strategy to be used by all components, whereas this ADR is focused on higher
+level details.
+
+## Why
+
+Project CodeFlare currently has different release processes and versioning schemes for all of its subprojects. This creates confusion for users when they wish to use the stack -- what components versions should they be using? How do external dependencies like KubeRay factor in?
+There are also no standards around testing for the project.
+This ADR aims to document the overall release process for Project CodeFlare.
+
+## Goals
+
+* Establish a common versioning scheme for all CodeFlare components
+* Establish a release process for Project CodeFlare
+* Establish guidelines for CodeFlare support matrix
+
+## Non-Goals
+
+* Require all components to restart versioning from scratch
+* Outline technical details for tests or CI infrastructure.
+
+## How
+
+We are proposing a unified release cycle for Project CodeFlare.
+We will release a new version of Project CodeFlare after every sprint (currently every 3 weeks), following normal semantic versioning. All components under the CodeFlare umbrella are expected to start using semantic versioning for future releases. Components are not expected to start versioning again from scratch (i.e v0.0.1). As part of a new release for Project CodeFlare, we will update the support matrix which outlines specific component versions and versions for important dependencies such as KubeRay. The components to be documented are as follows:
+
+* CodeFlare Operator
+  * The CodeFlare Operator version will be aligned with (i.e equivalent to) the Project CodeFlare version
+* CodeFlare SDK
+  * The CodeFlare Notebook Image will follow the same versioning.
+  * Supported TorchX version will be documented under the SDK.
+* InstaScale
+* MCAD
+* KubeRay
+  * On a new KubeRay release, CodeFlare will update it's supported version by the end of the next CodeFlare sprint
+
+### Images
+
+For published images, we are proposing repositories include the following three tags:
+
+1. Stable – This will be updated once every sprint as part of an official release
```

Comment:
I think it can be either.
My preference would be to create a new semver that becomes the stable release, but it is possible that there would be no difference between the previously released semver and a new one. In those cases, it doesn't make sense to generate a new release/update `stable`.